timesketch/lib/ts_sigma.py

```diff
@@ -0,0 +1,175 @@
+
+    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_file)
+
+    return sigma_config
+
+def get_sigma_rules_path():
+    """Get the path for Sigma rules.
+
+        Returns:
+            A string to the Sigma rules
+    """
+    # TODO: Add functionality to have multiple paths for rule files.
+    rules_path = current_app.config.get('SIGMA_RULES_FOLDER')
+
+    if not rules_path:
+        raise ValueError(
+            'SIGMA_RULES_FOLDER not found in config file')
+
+    if not os.path.isdir(rules_path):
+        raise ValueError(
+            'Unable to open dir: [{0:s}], it does not exist.'.format(
+                rules_path))
+
+    if not os.access(rules_path, os.R_OK):
+        raise ValueError(
+            'Unable to open dir: [{0:s}], cannot open it for '
+            'read, please check permissions.'.format(rules_path))
+
+    return rules_path
+
+
+def get_sigma_rules(rule_folder):
+    """Returns the Sigma rules for a folder including subfolders.
+
+        Returns:
+            A array of Sigma rules as JSON
+    """
+    return_array = []
+
+    for dirpath, dirnames, files in os.walk(rule_folder):
+        if 'deprecated' in [x.lower() for x in dirnames]:
+            dirnames.remove('deprecated')
+
+        for rule_filename in files:
+            if rule_filename.lower().endswith('.yml'):
+                # if a sub dir is found, do not try to parse it.
+                if os.path.isdir(os.path.join(dirpath, rule_filename)):
+                    continue
+
+                rule_file_path = os.path.join(dirpath, rule_filename)
+                parsed_rule = get_sigma_rule(rule_file_path)
+                return_array.append(parsed_rule)
+
+    return return_array
+
+
+def get_sigma_rule(filepath):
+    """ Returns a JSON represenation for a rule
+
+        Args:
+            filepath: path to the sigma rule to be parsed
+
+        Returns:
+            Json representation of the parsed rule
+        """
+    logger.info('get_sigma_rule with arg: {0:s}'.format(filepath))
+    sigma_config = get_sigma_config_file()
+
+    sigma_backend = sigma_es.ElasticsearchQuerystringBackend(sigma_config, {})
+
+    if filepath.lower().endswith('.yml'):
+        # if a sub dir is found, nothing can be parsed
+        if os.path.isdir(filepath):
+            return None
+
+        tag_name, _ = filepath.rsplit('.')
+        abs_path = os.path.abspath(filepath)
+
+        with codecs.open(
+                abs_path, 'r', encoding='utf-8', errors='replace') as file:
+            try:
+                rule_return = {}
+                rule_yaml_data = yaml.safe_load_all(file.read())
+                for doc in rule_yaml_data:
+                    rule_return.update(doc)
+                    parser = sigma_collection.SigmaCollectionParser(
+                        str(doc), sigma_config, None)
+                    parsed_sigma_rules = parser.generate(sigma_backend)
+
+            except NotImplementedError as exception:
```

Comment:
@kiddi please have a particular look at the cascading exceptions here (and the following lines) each of those occur when just going over the upstream sigma project rules folder. For sure I will file bugs for each of them upstream, but in the meantime I do not want our code to break, so I just return None for those rules.

The question I have here would you be okay with these cascading except blocks, or rather have only the generic one?

Reply:
I'm fine with these, since each of these exceptions is generated based on different conditions, so this will help with the debug of the error... this is much better than a broad `except Exception`, since now you can do a specific error handling, and specific instructions, so this is better